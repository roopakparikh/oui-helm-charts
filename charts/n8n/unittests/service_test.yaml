suite: test service

templates:
  - service.yaml

release:
  name: n8n
  namespace: n8n
chart:
  version: 1.0.0
  appVersion: 1.0.0

tests:
  - it: should use different port number when we change it
    set:
      service:
        port: 80
    documentSelector:
      path: metadata.name
      value: n8n
    asserts:
      - equal:
          path: spec.ports[0].port
          value: 80
      - equal:
          path: spec.ports[0].targetPort
          value: http

  - it: should use different port name when we change it
    set:
      service:
        name: differentname
    documentSelector:
      path: metadata.name
      value: n8n
    asserts:
      - equal:
          path: spec.ports[0].name
          value: differentname

  - it: should create webhook service when db type is postgresdb and webhook.mode is queue
    set:
      db:
        type: postgresdb
      webhook:
        mode: queue
      service:
        name: n8n-webhook
        port: 8080
    documentSelector:
      path: metadata.name
      value: n8n-webhook
    asserts:
      - equal:
          path: spec.ports[0].name
          value: n8n-webhook
      - equal:
          path: spec.ports[0].port
          value: 8080
      - equal:
          path: spec.ports[0].targetPort
          value: n8n-webhook

  - it: should match snapshot of default values
    set:
      service:
        annotations:
          test-key: test-value
    asserts:
      - matchSnapshot: { }
