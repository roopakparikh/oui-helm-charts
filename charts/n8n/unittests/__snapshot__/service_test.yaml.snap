should match snapshot of default values:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        test-key: test-value
      labels:
        app.kubernetes.io/instance: n8n
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: n8n
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: n8n-1.0.0
      name: n8n
    spec:
      ports:
        - name: http
          port: 5678
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/instance: n8n
        app.kubernetes.io/name: n8n
      type: ClusterIP
